pkg/util/errutil/errors.go

```diff
@@ -41,6 +41,23 @@ func NewBase(reason StatusReason, msgID string, opts ...BaseOpt) Base {
 	return b
 }
 
+func ErrorFrom(e error) (*Error, error) {
```

Comment:
We'll need to add tests for this before merging 🙂

And on second thought, we could just call this `From`, `errutil.From(err)` seems fairly clear to me. Hm, and maybe it'd make more sense to return a bool for status instead of returning the error passed 🤔 
```suggestion
func From(e error) (*Error, bool) {
```